lambda/subarea/POST/index.js

```diff
@@ -20,40 +37,112 @@ exports.handler = async (event, context) => {
       return sendResponse(403, { msg: 'Unauthorized.' }, context);
     }
 
-    if (event?.queryStringParameters?.type === "config") {
-      return await handleConfig(body, context);
-    } else if (event?.queryStringParameters?.type === "activity") {
-      // Handle standard monthly entry for this orc::subAreaName::activity
-      return await handleActivity(body);
-    } else {
-      throw "Invalid request";
+    // check if fiscal year is locked
+    if (await checkFiscalYearLock(body)) {
+      return sendResponse(403, { msg: `This fiscal year has been locked against editing by the system administrator.` }, context)
+    }
+
+    // Disabling the ability to change config for now. 
+    // Request used to necessitate 'type = activity/config' as a queryParam (was future proofing).
+    // Queryparams no longer required. All info included in request body. 
+    // Refer to code prior to 2022-09-27 for handleConfig.
+
+    await verifyBody(body);
+
+    // check if record is locked
+    const unlocking = lock === false ? true : false;
+    const existingRecord = await getOne(`${body.subAreaId}::${body.activity}`, body.date);
+    if (existingRecord?.isLocked && !unlocking) {
+      return sendResponse(403, { msg: 'Record is locked.' });
```

Comment:
HTTP 409 is a better response code.  You'll need to update your UT's accordingly.

Reply:
Fixed in `89812d6`.